lib/ecto/adapters/myxql.ex

```diff
@@ -198,6 +198,19 @@ defmodule Ecto.Adapters.MyXQL do
         {:error, exit_to_exception(exit)}
     end
   end
+  
+  @impl Ecto.Adapter.Storage
+  def storage_status(opts) do
+    database = Keyword.fetch!(opts, :database) || raise ":database is nil in repository configuration"
+    opts = Keyword.put(opts, :database, "information_schema")
```

Comment:
Maybe we should have a module attribute with the name `@default_maintenance_database`?
Much like what was already done [in the `postgres` adapter](https://github.com/elixir-ecto/ecto_sql/pull/164/files#diff-c34643168b4a4ab5a0aa4d51635f45c6R172)

Reply:
Honestly, any approach is fine.

Reply:
I believe we have `@default_maintenance_database` because on Postgres you always need to choose a DB when connecting to a server. There's no such requirement on MySQL, it's fine to connect to a server without picking one.

Thus I think we can safely remove line 205.

Reply:
I don't think we should remove 205 because we would try to connect to a database that may not yet exist? 🤔

Reply:
it's fine to connect to a mysql server without picking a database :)

Reply:
oh, I think I see your point, because we already have `opts[:database]` that's what we'll try to connect to. Yeah, I think this line is fine. We could set the database to `nil` too.

Reply:
let's mirror [`storage_up/1`](https://github.com/elixir-ecto/ecto_sql/blob/ee02e79166a238c4d73b2de734f762432daf929d/lib/ecto/adapters/myxql.ex#L160) where we drop the :database key.

Reply:
the "delete `:database` from options" is even done in the [`storage_up/1` function](https://github.com/elixir-ecto/ecto_sql/blob/fcbf61e06322bf3d6cdd2497edd705689351934c/lib/ecto/adapters/myxql.ex#L160), I think we are good to go  😄

Reply:
Fixed [here](https://github.com/elixir-ecto/ecto_sql/compare/ee02e79166a238c4d73b2de734f762432daf929d..aab4f7a628c00542cc9bff036a11ae56b1a86abb)